### DISCUSSION TITLE
Saml: dots in usernames still/again unsupported? What to do about them?

### DISCUSSION THREAD

**Discussion by verymilan** [mastodon/mastodon#17223] [Q&A]
Hi there, i wanted to connect an existing Keycloak instance using SAML. Initially it worked wonderfully - until a user with a dot in their username (`user.name123`) has tried to connect.
The history of this feature feels a bit chaotic and from the current regex, i am not sure wether it is allowed or forbidden. 

In any case, i would also be happy about some sort of workaround. @Gargron suggested getting rid of the dot in the past, but i am not sure what i could do to make it work.

**Comment by ClearlyClaire** [comment]
Dots are disallowed for local users for various reasons, so you'd have to somehow get rid of the dot for it to work. There was talk in a PR of automatically removing dots, but that leaves you open to collision issues (e.g. `user.name123` and `username123` both existing on your Keycloak instance)
